stl/inc/xhash

```diff
@@ -1915,6 +1915,169 @@ protected:
         return _List._Getal();
     }
 
+    struct _Multi_equal_check_result {
+        bool _Equal_possible;
+        _Unchecked_const_iterator _Subsequent_first; // only useful if _Equal_possible
+    };
+
+    _NODISCARD _Multi_equal_check_result _Multi_equal_check_equal_range(
+        const _Hash& _Right, _Unchecked_const_iterator _First1) const {
+        // check equal_range of elements matching *_First1 match the equal_range of elements in _Right
+        auto& _Keyval = _Traits::_Kfn(*_First1);
+        // find the start of the matching run in the other container
+        const size_t _Hashval   = _Right._Traitsobj(_Keyval);
+        const size_type _Bucket = _Hashval & _Right._Mask;
+        auto _First2            = _Right._Vec._Mypair._Myval2._Myfirst[_Bucket << 1];
+        if (_First2 == _Right._Unchecked_end()) {
+            // no matching bucket, therefore no matching run
+            return {false, {}};
+        }
+
+        const auto _Bucket_hi = _Right._Vec._Mypair._Myval2._Myfirst[(_Bucket << 1) + 1];
+        for (; _Right._Traitsobj(_Traits::_Kfn(*_First2), _Keyval); ++_First2) {
+            // find first matching element in _Right
+            if (_First2 == _Bucket_hi) {
+                return {false, {}};
+            }
+        }
+
+        if
+            _CONSTEXPR_IF(_Traits::_Standard) {
+                // trim matching prefixes
+                while (*_First1 == *_First2) {
+                    // the right equal_range ends at the end of the bucket or on the first nonequal element
+                    bool _Right_range_end = _First2 == _Bucket_hi;
+                    ++_First2;
+                    if (!_Right_range_end) {
+                        _Right_range_end = _Right._Traitsobj(_Keyval, _Traits::_Kfn(*_First2));
+                    }
+
+                    // the left equal_range ends at the end of the container or on the first nonequal element
+                    ++_First1;
+                    const bool _Left_range_end =
+                        _First1 == _Unchecked_end() || _Traitsobj(_Keyval, _Traits::_Kfn(*_First1));
+
+                    if (_Left_range_end && _Right_range_end) {
+                        // the equal_ranges were completely equal
+                        return {true, _First1};
+                    }
+
+                    if (_Left_range_end || _Right_range_end) {
+                        // one equal_range is a prefix of the other; different lengths mean not equal though

```

Comment:
Missing a "," before "though", but honestly I think this could simply be:
```suggestion
                        // one equal_range is a prefix of the other; not equal
```

Reply:
or even `// one equal_range is a proper prefix of the other`